utils.py

```diff
@@ -503,6 +504,37 @@ def convert_to_str(string_to_convert):
         return string_to_convert.encode('utf-8')
     return string_to_convert
 
+def get_sorted_nodes(nodes):
+    """Returns the nodes in topological sorted order.
+
+    Args:
+        nodes: list(dict). Unorderd list of nodes.
+
+    Returns:
+        list(dict). The list of nodes in topological sorted order.
```

Comment:
Unless it's impossible -- this docstring should acknowledge that case, and specify what gets returned instead.